Sources/Bow/Typeclasses/Divide.swift

```diff
@@ -0,0 +1,25 @@
+/// Divide is a typeclass that models the divide part of divide and conquer.
+///
+/// Divide basically states: Given a Kind<F, A> and a Kind<F, B> and a way to turn C into a tuple of A and B it gives you a Kind<F, C>.
+///
+public protocol Divide: Contravariant {
+    /// divide takes two data-types of type Kind<Self, A> and Kind<Self, B> and produces a type of Kind<Self, C> when given
+    /// a function from (C) -> (A, B).
+    /// - Parameter fa: data of type Kind<Self, A>
+    /// - Parameter fb: data of type Kind<Self, B>
+    /// - Parameter f: a function to transform C into (A, B)
+    /// - Returns: Kind<Self, C>
+    static func divide<A, B, C>(_ fa: Kind<Self, A>, _ fb: Kind<Self, B>, _ f: (C) -> (A, B)) -> Kind<Self, C>
```

Comment:
@truizlop I'd appreciate if you could guide me to how to go about implementing this for `EitherK`

Reply:
You probably need to require that `F` and `G` have instances of `Divide`, but I think that still with that, `EitherK` may not be able to produce a sensible instance of `Divide`. Consider that we may have 4 possible cases:

- `fa: F<A>, fb: F<B>` = F.divide(fa, fb, f)
- `fa: G<A>, fb: F<B>` = ???
- `fa: F<A>, fb: G<B>` = ???
- `fa: G<A>, fb: G<B>` = G.divide(fa, fb, f)

As you can see, there is no way to mix up an `F<A>` with a `G<B>`, so probably we cannot implement `Divide` for `EitherK`.